File: ios/Theme/RetryScreen.swift
public class RetryScreen {
  private static let KEY: String = "retryScreen"
  private let style: Style
  private let target: NSDictionary?
  private let font: Font
  private let color: Color

  init(target: NSDictionary?) {
    self.style = Style()
    self.target = self.style.getTarget(target, key: RetryScreen.KEY)
    self.font = Font()
    self.color = Color()
  }

  private func getInt32(key: String) -> Int32 {
    let defaultValue: Int32 = -1
    if !self.style.exists(self.target, key: key) {
      return defaultValue
    }

    return (self.target?[key] ?? defaultValue) as! Int32
  }

  public func getImageBorderWidth() -> Int32 {
    return self.getInt32(key: "imageBorderWidth")
  }

  public func getImageBorderColor() -> UIColor {
    return self.color.getColor(self.target, key: "imageBorderColor")
  }
  
  public func getHeaderTextColor() -> UIColor {
    return self.color.getColor(self.target, key: "headerTextColor", defaultColor: "#000000")
  }
  
  public func getHeaderFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "headerFont")
  }

  public func getImageCornerRadius() -> Int32 {
    return self.getInt32(key: "imageCornerRadius")
  }

  public func getSubtextColor() -> UIColor {
    return self.color.getColor(self.target, key: "subtextColor", defaultColor: "#000000")
  }
  
  public func getSubtextFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "subtextFont")
  }

  public func getOvalStrokeColor() -> UIColor {
    return self.color.getColor(self.target, key: "ovalStrokeColor")
  }
}

File: android/src/main/java/com/azifacemobile/theme/RetryScreen.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

import javax.annotation.Nullable;

public class RetryScreen {
  private final Theme theme;
  private final JSONObject target;
  private final Font font;
  private final Color color;

  public RetryScreen(ReactApplicationContext context, JSONObject target) {
    this.theme = new Theme();
    this.font = new Font(context);
    this.color = new Color();

    this.target = this.theme.getTarget(target, "retryScreen");
  }

  private int getInt(String key) {
    final int defaultValue = -1;
    try {
      if (!this.theme.exists(this.target, key)) {
        return defaultValue;
      }

      return this.target.getInt(key);
    } catch (JSONException e) {
      return defaultValue;
    }
  }

  public int getImageBorderWidth() {
    return this.getInt("imageBorderWidth");
  }

  public int getImageBorderColor() {
    return this.color.getColor(this.target, "imageBorderColor");
  }

  public int getImageCornerRadius() {
    return this.getInt("imageCornerRadius");
  }

  public int getHeaderTextColor() {
    return this.color.getColor(this.target, "headerTextColor", "#000000");
  }

  @Nullable
  public Typeface getHeaderFont() {
    return this.font.getTypography(this.target, "headerFont");
  }

  public int getSubtextColor() {
    return this.color.getColor(this.target, "subtextColor", "#000000");
  }

  @Nullable
  public Typeface getSubtextFont() {
    return this.font.getTypography(this.target, "subtextFont");
  }

  public int getOvalStrokeColor() {
    return this.color.getColor(this.target, "ovalStrokeColor");
  }
}
